pytential/symbolic/mappers.py

```diff
@@ -552,7 +560,12 @@ def map_product(self, expr: p.Product):
         if len(nonconst) > 1:
             raise _DerivativeTakerUnsupoortedProductError(
                     "DerivativeTaker doesn't support products with "
-                    "more than one non-constant")
+                    "more than one non-constant. "
+                    "The following were recognized as non-constant: "
+                    f"{', '.join(str(nc) for nc in nonconst)}. "
+                    "If some of these a spatially constant, use sym.SpatialConstant "
```

Comment:
```suggestion
                    "If some of these a spatially constant, use sumpy.symbolic.SpatialConstant "
```
Not sure if it's better to point to the original `sumpy` class or to our re-export?

Reply:
I thought about that. At the end of the day, the re-export felt more "consistent" to me. While there are some parts of `sumpy` sticking out in the pytential interface (the kernels, mostly), I feel that I'd like to minimize that. (Kernel derivatives also have a pytential dialect that gets resolved to what sumpy wants.)